### PR TITLE
Implement EnvironmentVariablePasswordProvider

### PR DESCRIPTION
**Pull request by nishantmonu51**
Adds EnvironmentVariablePasswordProvider which is capable of pulling metadata store password from environment variable.

### CODE REVIEW

File: common/src/test/java/io/druid/metadata/EnvironmentVariablePasswordProviderTest.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.io.IOException;
+
+public class EnvironmentVariablePasswordProviderTest
+{
+  private static final ObjectMapper jsonMapper = new ObjectMapper();
```
Comment (gianm) [inline common/src/test/java/io/druid/metadata/EnvironmentVariablePasswordProviderTest.java]: `DefaultObjectMapper`?

File: common/src/main/java/io/druid/metadata/EnvironmentVariablePasswordProvider.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+public class EnvironmentVariablePasswordProvider implements PasswordProvider
+{
+
+  private final String passwordKey;
+
+  @JsonCreator
+  public EnvironmentVariablePasswordProvider(@JsonProperty("passwordKey") String passwordKey)
+  {
+    this.passwordKey = passwordKey;
```
Comment (gianm) [inline common/src/main/java/io/druid/metadata/EnvironmentVariablePasswordProvider.java]: `passwordVariable` maybe, I rarely see env vars referred to as "keys"

**Comment by gianm** [comment]
@nishantmonu51 looks useful - could you add docs please?

File: common/src/test/java/io/druid/metadata/EnvironmentVariablePasswordProviderTest.java (hunk 1 above)
Comment (drcrallen) [inline common/src/test/java/io/druid/metadata/EnvironmentVariablePasswordProviderTest.java]: For maintainability sake I agree with @gianm

Comment (nishantmonu51) [inline common/src/test/java/io/druid/metadata/EnvironmentVariablePasswordProviderTest.java]: DefaultObjectMapper is not visible in druid-common, other unit tests also use it.

**Comment by nishantmonu51** [comment]
added docs.

**Comment by gianm** [comment]
👍

**Comment by drcrallen** [comment]
I really like this. It also allows custom password providers to grab passwords in complex ways.

@gianm  / @nishantmonu51 how would you feel about just making this an environment variable provider? it uses passwordprovider so the name of EnvironmentVariablePasswordProvider is accurate, but can we change the field to just "variable" instead of "passwordVariable"?

I can see a scenario where you have an internal PasswordProvider that can take sub-PasswordProviders as parameters...  for example:


```json
{
  "type":"companyPassFetcher",
  "key": "s0m3_k34",
  "secret":{
    "type":"environment",
    "variable":"SOME_DRUID_SECRET"
  }
}
```

**Comment by gianm** [comment]
I'm on board with naming it `"variable"`

**Comment by nishantmonu51** [comment]
@gianm @drcrallen: sounds good, renamed to variable.

File: common/src/test/java/io/druid/metadata/EnvironmentVariablePasswordProviderTest.java (hunk 1 above)
Comment (drcrallen) [inline common/src/test/java/io/druid/metadata/EnvironmentVariablePasswordProviderTest.java]: doh

**Comment by drcrallen** [comment]
👍 after jenkins

**Comment by gianm** [comment]
👍  after TRAVIS

DOWN WITH JENKINS

**Pull request merged**